runtime/lua/vim/lsp.lua

```diff
@@ -810,23 +811,38 @@ nvim_command("autocmd VimLeavePre * lua vim.lsp._vim_exit_handler()")
 ---  - Map of client-id:request-id pairs for all successful requests.
 ---  - Function which can be used to cancel all the requests. You could instead
 ---    iterate all clients and call their `cancel_request()` methods.
-function lsp.buf_request(bufnr, method, params, callback)
+function lsp.buf_request(bufnr, method, params, callback, unsupported_callback)
```

Comment:
I do not really like passing this parameter. Does this allow overriding of what to do with the error? (it's missing documentation)

Reply:
If so, perhaps there is a better way to do this (for example, registering at the buffer level / client level / lsp level).

Reply:
Yes, the idea is to override what to do with the error, so we don't get an error say `prepareRename` is not supported or something like that. The way rename works is:

1. call prepareRename
2. if prepareRename returns null, abort the process
3. if prepareRename, returns a valid response, use that to control how rename is called
4. if prepareRename is not supported, fallback to the previous behavior of calling rename with 

I added a separate callback to properly identify (4) because the default behavior is to log an error message, and in this case the message is non-actionable, so we don't want to log it. I agree that adding a new callback isn't the best solution.

A better solution would be to not call `err_message()` from the library function [here](https://github.com/neovim/neovim/blob/554b21261ec0ad1eaac7afc21d9a1ba7fb7cabf1/runtime/lua/vim/lsp.lua#L494). What do you think?

We still need a way to signal to rename that no clients implement prepareRename though, so we could have rename be aware of all the clients and count the callbacks, or have a new overall callback from buf_request.

What do you mean by registering at the buffer level? It sounds like that could work too, as long as we can update that registry on_attach or something like that.